usecases/cluster/state.go

```diff
@@ -63,12 +63,14 @@ type Config struct {
 	// failures (down nodes) faster.
 	FastFailureDetection bool `json:"fastFailureDetection" yaml:"fastFailureDetection"`
 	// LocalHost flag enables running a multi-node setup with the same localhost and different ports
-	Localhost bool `json:"localhost" yaml:"localhost"`
-	// MaintenanceNodes is experimental. is a list of nodes (by Hostname) that are in maintenance mode
-	// (eg return an error for all data requests). We use a list here instead of a bool because it
-	// allows us to set the same config/env vars on all nodes to put a subset of them in maintenance
-	// mode. In addition, we may want to have the cluster nodes not in maintenance mode be aware of
-	// which nodes are in maintenance mode in the future.
+	Localhost            bool `json:"localhost" yaml:"localhost"`
+	MaintenanceNodesLock *sync.RWMutex
```

Comment:
Q: I'm failing to understand the need for 
1. Why it has to be in Config? (can it be in `cluster/state.go` instead?)
2. Why has to be pointer type (given it's usage is direct like `s.config.MaintenanceNodesLock` without copying)

Two rationale
1. `sync.RWMutex` is usable immediately without needing explicit initialize. e.g: `var rw sync.RWMutex` is zero value and completely usable. 
2. From (1), may be it make sense to have it in `cluster/state.go` directly as value type (not pointer type)?
3. If it's value type, it's more simple, we can ignore all the `nil` checks in your code. Taking advantage of Go's zero-value usability :)

may be I miss something?

Reply:
Good idea, thanks, done